### DISCUSSION TITLE
How does "ISTIO_KUBE_APP_PROBERS" work?

### DISCUSSION THREAD

**Discussion by wo-o** [istio/istio#44289] [Q&A]
As I did tcpdump for the service container health check though istio-proxy, I found out that there is no connection between istio-proxy and service container for the health check. 

```
ISTIO_KUBE_APP_PROBERS:  {
  "/app-health/service-container/livez": {
    "httpGet": {
      "path": "/status/",
      "port": 8000,
      "scheme": "HTTP"
    },
    "timeoutSeconds": 30
  },
  "/app-health/service-container/readyz": {
    "httpGet": {
      "path": "/status/",
      "port": 8000,
      "scheme": "HTTP"
    },
    "timeoutSeconds": 10
  }
}
``` 

With above, I am curious why istio-proxy doesn't check <service-container>:8000/status/ directly to the service container.
Can you please explain the principle how istio-proxy check the service container's check ??

**Comment by howardjohn** [comment]
https://istio.io/latest/docs/ops/configuration/mesh/app-health-check/ has some details. Basically, if we kept the probe as-is, it would go through the proxy. Often, the proxy requires mTLS; kubelet doesn't do mtls. It also may do thingsd you don't want like telemetry, etc.

Instead, we hijack the probe to another special port which we proxy in a special way to bypass. For the user, it should be unknown
